examples/platform/esp32/mode-support/static-supported-modes-manager.cpp

```diff
@@ -0,0 +1,187 @@
+#include "static-supported-modes-manager.h"
+#include "ESP32Config.h"
+#include <app/util/debug-printing.h>
+#include <app/util/ember-print.h>
+
+using namespace chip;
+using namespace chip::app::Clusters;
+using namespace chip::DeviceLayer::Internal;
+using namespace chip::app::Clusters::ModeSelect;
+using chip::Protocols::InteractionModel::Status;
+
+using ModeOptionStructType = Structs::ModeOptionStruct::Type;
+using SemanticTag          = Structs::SemanticTagStruct::Type;
+template <typename T>
+using List = app::DataModel::List<T>;
+
+StaticSupportedModesManager::ModeOptionStructType * StaticSupportedModesManager::modeOptionStructList = nullptr;
+StaticSupportedModesManager::SemanticTag * StaticSupportedModesManager::semanticTags                  = nullptr;
```

Comment:
These two variables are only used inside the `getModeOptionsProvider` function, except for `~StaticSupportedModesManager`.  Why do they need to be class statics?  They should be function-local, and `~StaticSupportedModesManager` should clean up unconditionally, I would think.

If we do keep the existing conditional cleanup, it would be safer to condition it on a boolean that gets set when we allocate our first mode, instead of exposing these pointers.

Reply:
Made these two variables function-local.